lib/Mongo/MongoClient.php

```diff
@@ -229,7 +262,10 @@ public function listDBs()
      */
     public function selectCollection($db, $collection)
     {
-        return new MongoCollection($this->selectDB($db), $collection);
+        $collection = new MongoCollection($this->selectDB($db), $collection);
+        $collection->setReadPreference($this->readPreference['type'], $this->readPreference['tagsets']);
```

Comment:
not 100% sure if I am getting what you are saying right, this could be OK?

```php
    public function __construct(MongoDB $db, $name, array $options = [])
    {
        $this->db = $db;
        $this->name = $name;
        $this->collection = $this->db->getDb()->selectCollection($name, $options);
    }  
```

`options` will contain the read preference and write concern with the format the mongodb userland lib expects it.

Does it looks OK?